compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -165,6 +165,8 @@ object Types {
       case _: SingletonType | NoPrefix => true
       case tp: RefinedOrRecType => tp.parent.isStable
       case tp: ExprType => tp.resultType.isStable
+      case AppliedType(tycon: TypeRef, args: List[Type])
```

Comment:
Very nervous about performance implications here. `isStable` is a hotspot and what you do here seems to be about an order of magnitude more expensive than the rest. If we want to go down that route we probably should cache the value in `AppliedType`.